arch/arc/core/regular_irq.S

```diff
@@ -255,12 +254,12 @@ SECTION_FUNC(TEXT, _rirq_exit)
 	push_s r3
 #endif
 
+	/* _thread_arch.relinquish_cause is 32 bit despite of platform bittnes */
+	_st32_huge_offset _CAUSE_RIRQ, r2, _thread_offset_to_relinquish_cause, r1
```

Comment:
I assume the use of r1 instead of r2 as in the original is intended?

Reply:
Indeed. Last argument is an register we use as scratch - so we can't use r2 here.